src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs

```diff
@@ -19,96 +23,119 @@ namespace Microsoft.Health.SqlServer.Registration
-
-            services.Add<SqlConnectionWrapperFactory>()
-                .Scoped()
-                .AsSelf()
-                .AsImplementedInterfaces();
-
-            services.AddSingleton<ISqlConnectionStringProvider, DefaultSqlConnectionStringProvider>();
-
-            switch (config.AuthenticationType)
-            {
-                case SqlServerAuthenticationType.ManagedIdentity:
-                    services.AddSingleton<ISqlConnectionFactory, ManagedIdentitySqlConnectionFactory>();
-                    services.AddSingleton<IAccessTokenHandler, ManagedIdentityAccessTokenHandler>();
-                    services.AddSingleton<AzureServiceTokenProvider>();
-                    break;
-                case SqlServerAuthenticationType.ConnectionString:
-                default:
-                    services.AddSingleton<ISqlConnectionFactory, DefaultSqlConnectionFactory>();
-                    break;
-            }
+            return services;
+        }
+
+        /// <summary>
+        ///  Adds a collection of services for connecting to SQL Server to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to be updated.</param>
+        /// <param name="configure">A delegate for configuring the <see cref="SqlServerDataStoreConfiguration"/>.</param>
+        /// <returns>The <paramref name="services"/> for additional method invocations.</returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="services"/> or <paramref name="configure"/> is <see langword="null"/>.
+        /// </exception>
+        public static IServiceCollection AddSqlServerConnection(
+            this IServiceCollection services,
+            Action<SqlServerDataStoreConfiguration> configure)
+        {
+            EnsureArg.IsNotNull(configure, nameof(configure));
+
+            return services
+                .AddSqlServerConnection()
+                .Configure(configure);
+        }
+
+        /// <summary>
+        ///  Adds a hosted service for updating the application's SQL database to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <typeparam name="TVersion">The type of the version enumeration.</typeparam>
+        /// <param name="services">The <see cref="IServiceCollection"/> to be updated.</param>
+        /// <returns>The <paramref name="services"/> for additional method invocations.</returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="services"/> is <see langword="null"/>.
+        /// </exception>
+        public static IServiceCollection AddSqlServerVersioningService<TVersion>(this IServiceCollection services)
+            where TVersion : Enum
+        {
+            EnsureArg.IsNotNull(services, nameof(services));
+
+            services.TryAddScoped<ISchemaDataStore, SqlServerSchemaDataStore>();
```

Comment:
For applications that wish to read the version without taking an action on it, I should also expose an `ISchemaDataStore` for use without the `IHostedService`